groups/forms.py

```diff
@@ -70,26 +70,15 @@ class DiscussionCreate(forms.Form):
     )
 
 
-class DiscussionSubscribeForm(forms.Form):
+class SubscribeFormMixin(object):
     subscribe = forms.BooleanField(widget=forms.HiddenInput(), required=False)
 
-    class Meta:
-        fields = ('subscribe',)
-
-    def __init__(self, user, discussion, *args, **kwargs):
-        """
-        Build the layout to reflect the action the form will take.
-
-        Accepts (and requires) a user and a discussion as keyword arguments.
-        """
-        to_subscribe = user not in discussion.subscribers.all()
-
-        # setdefault is like 'get', but if it misses, puts the specified
-        # default into kwargs and returns *that* object.
+    def __init__(self, user, instance, *args, **kwargs):
+        to_subscribe = self.to_subscribe(user, instance)
```

Comment:
Keep the docstring, please - it's important that libraries are easy for other programmers inside and outside Incuna to understand.